packages/modal/src/Modal.js

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect } from "react";
+import ReactDOM from "react-dom";
+import PropTypes from "prop-types";
+import { useTransition, animated } from "react-spring";
+import Link from "@crave/farmblocks-link";
+
+import {
+  Wrapper,
+  Overlay,
+  ConstrainedCard,
+  ContentWrapper,
+  Header,
+} from "./Modal.styled";
+
+const Modal = ({
+  isOpen,
+  parentNode,
+  shouldCloseOnOverlayClick,
+  shouldCloseOnEsc,
+  onRequestClose,
+  showCloseButton,
+  children,
+  cardProps,
+  closeProps,
+  className,
+}) => {
+  if (!parentNode) return null;
+
+  const handleKeyDown = event => {
+    if (shouldCloseOnEsc && event.key === "Escape") {
+      onRequestClose?.(event);
+    }
+  };
+
+  // eslint-disable-next-line consistent-return
+  useEffect(() => {
+    if (isOpen) {
+      const originalOverflow = parentNode.style?.overflow;
+
+      // eslint-disable-next-line no-param-reassign
+      parentNode.style.overflow = "hidden";
+      parentNode.addEventListener("keydown", handleKeyDown);
+      return () => {
+        // eslint-disable-next-line no-param-reassign
+        parentNode.style.overflow = originalOverflow || "";
+        parentNode.removeEventListener("keydown", handleKeyDown);
+      };
+    }
+  }, [isOpen]);
+
+  const fade = useTransition(isOpen, null, {
+    from: {
+      opacity: 0,
+      position:
+        parentNode === Modal.defaultProps.parentNode ? "fixed" : "absolute",
+    },
+    enter: { opacity: 1 },
+    leave: { opacity: 0 },
+  });
+  const slide = useTransition(isOpen, null, {
+    from: { transform: "translate3D(0, 50px, 0)", maxHeight: "100%" },
+    enter: { transform: "translate3D(0, 0px, 0)" },
+    leave: { transform: "translate3D(0, -100px, 0)" },
+    unique: true,
+  });
+
+  return ReactDOM.createPortal(
+    fade.map(
+      ({ item: fadeItem, key: fadeKey, props: fadeStyle }) =>
+        fadeItem && (
+          <Wrapper className={className} key={fadeKey} style={fadeStyle}>
+            <Overlay
+              data-testid="modal-overlay"
+              className="overlay"
+              onClick={shouldCloseOnOverlayClick ? onRequestClose : undefined}
+            />
+            {slide.map(
+              ({ item: slideItem, key: slideKey, props: slideStyle }) =>
+                slideItem && (
+                  <animated.div key={slideKey} style={slideStyle}>
+                    <ConstrainedCard floating className="card" {...cardProps}>
+                      {showCloseButton && (
+                        <Header className="header">
+                          <Link
+                            className="close"
+                            rightIcon="wg-close-int"
```

Comment:
It looks like something is cutting the icon... IDK if it's here on Modal or on Link

![image](https://user-images.githubusercontent.com/17936244/58588713-a33c6580-8236-11e9-9d76-16bb5ffc57b8.png)

Reply:
It's in the icons font

Reply:
Right... So we should either update this icon on our fonticon or use another icon here